spyder/widgets/tests/test_status.py

```diff
@@ -40,5 +42,51 @@ def test_status_bar(status_bar, qtbot):
     assert len(swidgets) == 2
 
 
+class StatusBarWidgetTest(StatusBarWidget):
+    def get_tooltip(self):
+        return 'tooltip'
+
+    def get_icon(self):
+        return 'icon'
+
+
+def test_status_bar_widget_signal(status_bar, qtbot):
+    win, statusbar = status_bar
+    w = StatusBarWidgetTest(win, statusbar)
+
+    with qtbot.waitSignal(w.sig_clicked, timeout=1000):
+        qtbot.mouseRelease(w, Qt.LeftButton)
+
+    assert w.get_tooltip() == 'tooltip'
+    assert w.get_icon() == 'icon'
+
+
+def test_status_bar_conda_status(status_bar, qtbot, mocker):
```

Comment:
This is failing on Windows. Since the test seem to be adapted to Posix, you should skip it or rewrite to work on both Windows and Posix.

Reply:
I generalized the code to make it work on all platforms.

Reply:
Nice, thanks!